subprojects/base-services/src/main/java/org/gradle/util/GFileUtils.java

```diff
@@ -69,7 +71,26 @@ public static void touchExisting(File file) {
         try {
             Files.setLastModifiedTime(file.toPath(), FileTime.fromMillis(System.currentTimeMillis()));
         } catch (IOException e) {
-            throw new UncheckedIOException("Could not update time stamp for " + file, e);
+            if (file.isFile() && file.length() == 0) {
```

Comment:
I think we can avoid the special `file.length() == 0` case by using the `append` parameter:
https://stackoverflow.com/a/16927309

Reply:
I tried that but writing an empty byte array using `append = true` does not change the file's modification timestamp (neither on macOS and nor in the Docker container you've provided).